### DISCUSSION TITLE
Microsteps suggestions.

### DISCUSSION THREAD

**Discussion by vegetate7** [kachurovskiy/nanoels#175] [Q&A] [answered]
Hello! While experimenting with DRO, I faced this issue:
![IMG_20240203_202404](https://github.com/kachurovskiy/nanoels/assets/57148620/6f07d16e-adfd-47c1-a5d1-532ea91ae86d)
Look at the 1um difference. It's introduced by the Z screw pitch 1.5mm (15000du) which is coefficient of 3. Stepper driver set to 1600 microsteps. So divide and round() can make this effect.
Should I set driver to 3600 (which is minimal setting having 3 in multipliers)? Can NanoELS work with such small steps? It will lower max RPM and/or pitch.
May be exists any general rule to select microsteps according to screw pitch?

**Comment by kachurovskiy** [comment, marked as answer]
> Should I set driver to 3600 (which is minimal setting having 3 in multipliers)? Can NanoELS work with such small steps? It will lower max RPM and/or pitch.

Yes, H4 can work with small steps but you'd be reducing your stepper motor maximum reachable RPM by 2x with every micro-step increase. High speed axis movement is not normally needed but with such high micro-stepping it might start to affect your everyday work.

I normally use 800 steps on the drivers as there are benefits of using micro-stepping beyond that are diminishing and drawbacks are getting more pronounced.

There should be no accumulating division and rounding errors in H4 since we're dealing with longs (encoder and motor step count). It is still possible to get non-round numbers on display in inches or millimeters due to thread and screw pitches as you mentioned. Hopefully it doesn't disturb the working process too much :)

**Reply by vegetate7** [reply]
800 was too loud for me on less than 0.2mm pitch and ~250rpm, that's why I moved to 1600.

Ok, I got it. Thanks!
